app/Service/CollaborationService.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Models\Student;
+use App\Models\Collaboration;
+
+class CollaborationService
+{
+    public function getCollaborationDetails($uuid)
+    {
+        $student = Student::where('id', $uuid)->with('resume')->firstOrFail();
+        $resume = $student->resume;
+        $collaborationIds = json_decode($resume->collaborations_ids);
+        $collaborations = Collaboration::findMany($collaborationIds);
+
+        return $collaborations->map(function ($collaboration) {
+            return [
+                'uuid' => $collaboration->id,
+                'collaboration_name' => $collaboration->collaboration_name,
+                'collaboration_description' => $collaboration->collaboration_description,
+                'collaboration_quantity' => $collaboration->collaboration_quantity,
+            ];
+        });
+    }
```

Comment:
According to the suggestion made in the controller, the service is the place from which the different exceptions are thrown and later caught in the controller.

As a suggestion, you could apply this modification, which also includes an idea that @coxen34 applied in a previous PR to extract the functionality of retrieving details into a separate function:

~~~
public function execute($studentId)
    {
        return $this->getCollaborationDetails($studentId);
    }

    public function getCollaborationDetails($studentId)
    {
        $student = Student::find($studentId);

        if (!$student) {
            throw new StudentNotFoundException($studentId);
        }

        $resume = $student->resume()->first();

        if (!$resume) {
            throw new ResumeNotFoundException($studentId);
        }

        $collaborationIds = json_decode($resume->collaborations_ids);
        
        $collaborations = Collaboration::findMany($collaborationIds);

        return $collaborations->map(function ($collaboration) {
            return [
                'uuid' => $collaboration->id,
                'collaboration_name' => $collaboration->collaboration_name,
                'collaboration_description' => $collaboration->collaboration_description,
                'collaboration_quantity' => $collaboration->collaboration_quantity,
            ];
        });
    }
~~~

**Important**: if you decide to apply these changes, you must make a small modification to the function called ``public function it_throws_model_not_found_exception_for_invalid_uuid()`` in the service test file.